public/app/features/explore/RichHistory/RichHistoryAddToLibrary.tsx

```diff
@@ -0,0 +1,43 @@
+import { t } from 'i18next';
+import React from 'react';
+
+import { AppEvents } from '@grafana/data';
+import { getAppEvents } from '@grafana/runtime';
+import { DataQuery } from '@grafana/schema';
+import { Button } from '@grafana/ui';
+import { isQueryLibraryEnabled, useAddQueryTemplateMutation } from 'app/features/query-library';
+import { AddQueryTemplateCommand } from 'app/features/query-library/types';
+
+type Props = {
+  query: DataQuery;
+};
+
+export const RichHistoryAddToLibrary = ({ query }: Props) => {
+  const [addQueryTemplate, { isSuccess }] = useAddQueryTemplateMutation();
+
+  const handleAddQueryTemplate = async (addQueryTemplateCommand: AddQueryTemplateCommand) => {
+    const result = await addQueryTemplate(addQueryTemplateCommand);
+    if (!result.error) {
+      getAppEvents().publish({
+        type: AppEvents.alertSuccess.name,
+        payload: [
+          t('explore.rich-history-card.query-template-added', 'Query template successfully added to the library'),
+        ],
+      });
+    }
+  };
+
+  const buttonLabel = t('explore.rich-history-card.add-to-library', 'Add to library');
+
+  return isQueryLibraryEnabled() && !isSuccess ? (
+    <Button
+      variant="secondary"
+      aria-label={buttonLabel}
+      onClick={() => {
+        handleAddQueryTemplate({ title: 'Test', targets: [query] });
```

Comment:
I know this will get replaced later, but I'd like the title field to be optional, could we have a better default here? Like "Imported from Explore - {timestamp}" or something?

Reply:
> but I'd like the title field to be optional,

By optional you mean so the user can leave the input blank and then it'd default to "Imported from Explore - {timestamp}"? 

cc @diegoadams. Not sure if we've discussed. In your designs - is the description field required or would default to something it the description is not provided? If we fallback to a default, what's the best way to show it to the user? With a placeholder in the input field?

Reply:
Added in db9a12f8482d32a6f02362b84f4aa51b588576b2

Reply:
Great! Already approved but now I EXTRA approve :D